src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -50,31 +45,31 @@ public void Dispose()
         [System.Security.SecuritySafeCritical]  // auto-generated
         private unsafe void Dispose(bool disposing)
         {
-            if (_store != null)
+            if (_stream != null)
             {
 
                 if (disposing)
                 {
                     // Close the stream in a thread-safe way.  This fix means 
                     // that we may call Close n times, but that's safe.
-                    BinaryReader copyOfStore = _store;
-                    _store = null;
-                    if (copyOfStore != null)
-                        copyOfStore.Dispose();
+                    Stream stream = _stream;
```

Comment:
Hmm, I'm not sure about that.  I took a quick look just now, and it looks like the locks are used at or close to the top level of entrypoints into an instance, so while not every member is using a lock, those members appear to only be called while a lock is held.  And since there are locks, curent code could easily be relying on that, even though the type is documented such that members aren't thread-safe.  My recommendation would be to:
- Remove this comment. It's not correct, and Dispose should only be done when no one else is using the instance.
- Add ```Debug.Assert(Monitor.IsEntered(this))``` to the beginning of methods that access the stream without a lock, either helping to ensure that they're used while the lock is held or showing that I'm wrong and there are cases where they're called without the lock held.